sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ExpressionParserSuite.scala

```diff
@@ -179,6 +179,40 @@ class ExpressionParserSuite extends PlanTest {
       InSubquery(Seq('a), ListQuery(table("c").select('b))))
   }
 
+  test("any sub-query") {
+    // Since the `genCmp` is generated from `AstBuilder` as a lambda expression
+    // which is not equal to the original comparison expression like `EqualTo`,
+    // we need to extract it from the parsed expression.
+    def extractPredicate(expr: Expression): (Expression, Expression) => Expression = {
+      expr.asInstanceOf[AnySubquery].genCmp
+    }
+
+    val any_subquery = defaultParser.parseExpression("a < any (select b from c)")
+    compareExpressions(
+      any_subquery,
+      AnySubquery(Seq('a), ListQuery(table("c").select('b)), extractPredicate(any_subquery))
+    )
+
+    val some_subquery = defaultParser.parseExpression("a = some (select b from c)")
+    compareExpressions(
+      some_subquery,
+      AnySubquery(Seq('a), ListQuery(table("c").select('b)), extractPredicate(some_subquery))
+    )
```

Comment:
I think we don't need `Some` tests here.

Reply:
ok